frontend/packages/dev-console/src/components/monitoring/alerts/monitoring-alerts-utils.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { expandable, sortable, cellWidth } from '@patternfly/react-table';
+import { StateCounts, Severity, AlertState } from '@console/internal/components/monitoring';
+import { Kebab } from '@console/internal/components/utils';
+import {
+  alertDescription,
+  alertState,
+  AlertStates,
+  AlertSeverity,
+} from '@console/internal/reducers/monitoring';
+import { Alert, Rule } from '@console/internal/components/monitoring/types';
+
+const viewAlertRule = {
+  label: 'View Alerting Rule',
+};
+
+const alertStateReducer = (alert: Alert) => _.toLower(alert?.state);
+const alertSeverityReducer = (alert: Alert) => alert?.labels?.severity;
+
+const alertStates = [AlertStates.Firing, AlertStates.Pending, AlertStates.Silenced];
+
+const alertSeverities = [AlertSeverity.Critical, AlertSeverity.Warning, AlertSeverity.Info];
+
+export const monitoringAlertColumn = [
+  {
+    title: 'Name',
+    cellFormatters: [expandable],
+    transforms: [sortable],
+  },
+  { title: 'Severity', transforms: [sortable, cellWidth(20)] },
+  { title: 'Alert State', transforms: [sortable, cellWidth(20)] },
+  '',
+];
+
+export const monitoringAlertRows = (alertrules: Rule[]) => {
+  const rows = [];
+  _.map(alertrules, (rls) => {
+    rls.type !== 'recording' &&
+      rows.push({
+        isOpen: rls.state === 'FIRING',
+        cells: [
+          rls.name,
+          {
+            title: <Severity severity={rls.labels?.severity} />,
+          },
+          {
+            title: <StateCounts alerts={rls.alerts} />,
+          },
+          {
+            title: (
+              <div className="odc-monitoring-alerts--kebab">
+                <Kebab options={[viewAlertRule]} />
+              </div>
+            ),
+          },
+        ],
+      });
+    rls.alerts &&
+      _.map(rls.alerts, (alert: Alert) => {
+        rows.push({
+          parent: _.findIndex(rows, (r) => r.cells[0] === rls.name),
+          fullWidth: true,
+          cells: [
+            {
+              title: (
+                <div className="odc-monitoring-alerts--alert-title">{alertDescription(alert)}</div>
+              ),
+              props: { colSpan: 3 },
+            },
+            {
+              title: (
+                <div className="odc-monitoring-alerts--state-column">
+                  <AlertState state={alertState(alert)} />
+                </div>
+              ),
+              props: { colSpan: 2 },
+            },
+          ],
+        });
+      });
+  });
+  return rows;
+};
+
+export const alertRowFilters = [
```

Comment:
https://github.com/openshift/console/pull/5527 is adding `alertsRowFilter`, so we should be able to use that instead of defining it here.

We want to add a severity filter to the admin perspective too, so we can move that from here to the shared `alertsRowFilter`.

Reply:
Sure,  I will reuse it from the console once PR #5527 gets in.

Reply:
https://github.com/openshift/console/pull/5527 has now merged

Reply:
Updated it.